src/VkBootstrap.cpp

```diff
@@ -320,9 +467,27 @@ const char* to_string (SwapchainError err) {
 	}
 }
 
+bool init_vulkan_funcs (PFN_vkGetInstanceProcAddr fp_vkGetInstanceProcAddr) {
+	if (!detail::funcs.load_vulkan (fp_vkGetInstanceProcAddr)) return false;
+	detail::funcs.init_pre_instance_funcs ();
```

Comment:
This is not thread-safe. If it is intended to requery the function pointers, then maybe it would be better to store the pointers inside the SystemInfo. Otherwise using function statics or a mutex would make it threadsafe.

Reply:
I didn't see a clear way to use function statics that would allow for data sharing between the initialization and getting of function pointers. Instead I opted to use mutexes, which should solve the problem readily, if at the cost of needing to use the `<mutex>` header. But well, even if users are very unlikely to do this, it's better to be safe than sorry.